cloudify_azure/resources/base.py

```diff
@@ -362,12 +362,35 @@ def delete(self, name):
                 retry_after=self.get_retry_after(headers))
         # HTTP 400 (BAD_REQUEST) - We're sending bad data
         elif res.status_code == httplib.BAD_REQUEST:
+            # Sometime it takes longer time to fully delete a parent node, so
+            # lets wait for sometime to give it a chance to be fully deleted
+            # This check will cover: [NicInUse, InUseSubnetCannotBeDeleted,
+            # InUseNetworkSecurityGroupCannotBeDeleted]
+            if "InUse" in res.json()['error']['code']:
+                self.log.warn(
+                    "The resource currently in use, maybe a deletion for a "
+                    "parent node using it still in progress, let's wait: {0}"
+                        .format(headers)
+                )
+                raise RecoverableError('Resource in use')
             self.log.info('BAD REQUEST: response: {}'.format(
                 utils.secure_logging_content(res.content)))
             raise UnexpectedResponse(
                 'Recieved HTTP 400 BAD REQUEST', res.json())
         # HTTP 409 (CONFLICT) - Operation failed
         elif res.status_code == httplib.CONFLICT:
+            # Sometime it takes longer time to fully delete a parent node, so
+            # lets wait for sometime to give it a chance to be fully deleted
+            # Checking against message content as codes are not informative
+            # [OperationNotAllowed, AccountIsLocked]
+            if "ensure that it does not contain any VM" in res.text or \
+                    "its artifacts being in use" in res.text:
```

Comment:
Is this the best identifier in the API response?

Reply:
In my testing, I've faced two `409` errors due to resource is still being in use:

1. 
```
"error": {
    "message": "The storage account cannot be deleted due to its artifacts being in use. For more information on troubleshooting this issue, see https://azure.microsoft.com/documentation/articles/storage-cannot-delete-storage-account-container-vhd/.",
    "code": "AccountIsLocked"
  }
```

2. 
```
"error":  {
"message": "Availability Set 'cfyghmdbavailset1' cannot be deleted. Before deleting an Availability Set please ensure that it does not contain any VM.",
"code": "OperationNotAllowed"
}
```
As you can see, nothing standard here -- so this is the best thing I could use for identifying these errors

Reply:
Cannot we depend on the code ? and since it is a 409 status code that means the resources is still exists so why do we just stick with the 409 status code

Reply:
hmmmm, well, honestly I don't know the other scenarios which is marked as NonRecoverable, so I'd like to update what I need only -- just the ones I'm sure that retry might fix it